### DISCUSSION TITLE
Why are there 3 traces in one request?

### DISCUSSION THREAD

**Discussion by hgaong** [apache/skywalking#11368] [Q&A]
Why are there 3 traces in one request?
<img width="1116" alt="image" src="https://github.com/apache/skywalking/assets/29444991/4126b902-8b00-4f30-bf6d-45c168b58b24">

--------------------

Why does the second trace appear separately instead of being part of the local type trace chain below it?
<img width="958" alt="image" src="https://github.com/apache/skywalking/assets/29444991/278a963c-9d45-4fca-8fc1-c95045c455f5">

<img width="555" alt="image" src="https://github.com/apache/skywalking/assets/29444991/621a9ca2-8e8d-413c-b0b2-fbd219096d67">
The tagUserId is invoked in the AuthFilter, which implements GlobalFilter.

**Comment by wu-sheng** [comment]
The left list is trace segments, not a trace or 3 traces.

**Comment by hgaong** [comment]
Why are there 3 segments? In a single request, I only want one segment, and the first and third segments are completely identical.



---Original---
From: "吴晟 Wu ***@***.***&gt;
Date: Tue, Oct 3, 2023 07:07 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [apache/skywalking] Why are there 3 traces in one request?(Discussion #11368)




 
The left list is trace segments, not a trace or 3 traces.

**Reply by wu-sheng** [reply]
Trace segment is determined by the agent, how to collect spans.
In your screenshot, it is clearly there are 3 services involved in a request path, so at least 3 segments from 3 services.

**Comment by hgaong** [comment]
Look at the content within the green box in the first image. What I'm asking is why there are 3&nbsp; seg (or something else) in the left-hand list. The second one in the left-hand list is generated by @trace. You mentioned that there should be at least three services, but without @trace, there are only two. How can this be explained?



---Original---
From: "吴晟 Wu ***@***.***&gt;
Date: Tue, Oct 3, 2023 08:07 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [apache/skywalking] Why are there 3 traces in one request?(Discussion #11368)




 
Trace segment is determined by the agent, how to collect spans.
 In your screenshot, it is clearly there are 3 services involved in a request path, so at least 3 segments from 3 services.

**Reply by wu-sheng** [reply]
I don't have time to explain your details. How your codes run is your env. I gave your explanation about what trace segment is.

**Reply by wu-sheng** [reply]
The second part of your question had been answered here, https://github.com/apache/skywalking/discussions/11367#discussioncomment-7165578. It means trace broken.
